### PR TITLE
Support for sacc reading and writing + cosmic shear example notebook and likelihood

### PR DESCRIPTION
**Pull request by arthurmloureiro**
This MR adds the following features:
- cosmic shear example likelihood
- new infrastructure to take sacc objects to be blinded into their likelihoods
- ability to save the blinded data-vector in sacc format
- new unit tests for new functions

API Changes:
- The Blinding Module now requires a sacc object.

### CODE REVIEW

**Comment by arthurmloureiro** [comment]
Testing is breaking because "save_to_sacc" is a new firecrown feature and possibly not installable via the environment.yaml

File: examples/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Examples
+This directory contains examples on how to build a firecrown likelihood that could be used to to blind datavectors. 
+
+These need to contain the following methods:
+- `build_likelihood` which returns a `firecrown.Likelihood` object
```
Comment (tilmantroester) [inline examples/README.md:5]: and `ModellingTools`?

File: examples/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+- `.compute_theory_vector(ModellingTools)` which returns the theory vector for the given model
+- `.get_data_vector()` which returns the data vector from the likelihood
+
```
Comment (tilmantroester) [inline examples/README.md:8]: In principle this shouldn't be necessary if you're already assuming a firecrown likelihood. At least once these methods have been defined in a base class.

File: examples/cosmic_shear/cosmicshear_sacc.fits
[the remark this answers is not in the document]
Comment (arthurmloureiro) [reply examples/cosmic_shear/cosmicshear_sacc.fits:1]: It is really minimal: 37 KB.

File: src/blinding/smokescreen.py [hunk 3]
```diff
@@ -196,16 +233,17 @@ def calculate_blinding_factor(self, type="add"):
 
         # update the likelihood with the systematics parameters:
         self.likelihood.update(self.systematics)
+        self._bld_likelihood.update(self.systematics)
```
Comment (tilmantroester) [inline src/blinding/smokescreen.py:236]: Why do you need a copy of the likelihood and tools when they both presumably read the same sacc file?

Comment (arthurmloureiro) [reply src/blinding/smokescreen.py:236]: I need to call two theory vectors: one with a fiducial cosmology and one with a shifted cosmology.

Firecrown safeguards don't allow me to call a theory vector calculation twice.

Comment (arthurmloureiro) [reply src/blinding/smokescreen.py]: Fixed by [36d45f8](https://github.com/LSSTDESC/Blinding/commit/36d45f80058d837223e7f3aceb7b8704506e8ebb)

**Pull request merged**